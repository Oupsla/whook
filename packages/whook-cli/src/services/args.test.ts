import initArgs from './args';

describe('initArgs', () => {
  const log = jest.fn();

  beforeEach(() => {
    log.mockReset();
  });

  it('should parse args', async () => {
    const args = await initArgs({
      log,
      ARGS: ['handler', '--name', 'getPing', '--parameters', '{}'],
    });

    expect({
      logCalls: log.mock.calls.filter(args => 'stack' !== args[0]),
      args,
    }).toMatchSnapshot();
  });
});
