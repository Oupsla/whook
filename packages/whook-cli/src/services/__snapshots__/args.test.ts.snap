// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initArgs should parse args 1`] = `
Object {
  "args": Object {
    "_": Array [
      "getPing",
    ],
    "parameters": "{}",
  },
  "logCalls": Array [
    Array [
      "debug",
      "ðŸ›  - Parsed args:",
      Object {
        "_": Array [
          "getPing",
        ],
        "parameters": "{}",
      },
    ],
  ],
}
`;
